src/components/Cohorts/CohortsTopBar/LoadingButton.tsx

```diff
@@ -6,17 +6,23 @@ import { MembersCountBox } from '../../Members/CohortMembersTopBar/style';
 interface IProps {
   btnTitle: string;
   btnStartIcon: React.ReactElement;
+  cohortsCount: string;
+  handleClick: () => void;
 }
 
-const LoadingBtn = ({ btnTitle, btnStartIcon }: IProps) => {
+const LoadingBtn = ({ btnTitle, btnStartIcon, cohortsCount, handleClick }: IProps) => {
   const [loading, setLoading] = React.useState(true);
-  function handleClick() {
-    setLoading(true);
-  }
 
-  setTimeout(() => {
-    setLoading(false);
-  }, 3000);
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      setLoading(false);
+    }, 3000);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, []);
```

Comment:
Clear the timer here and use `isLoading` from `react-query` you can pass it as props

Reply:
You delete `loading` property but I said to pass it as a props from react query